src/uucore/src/lib/mods/line_ending.rs

```diff
@@ -0,0 +1,42 @@
+/// Aims to provide consistent newline/zero terminator handling
+use std::fmt::Display;
+
+#[repr(u8)]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+/// Line terminator used for printing and parsing
+pub enum LineEnding {
+    #[default] Newline = b'\n',
+    Nul = 0,
+    Space = b' ',
+    None = 8, // abuse backspace \b to encode None
```

Comment:
This feels like premature optimization to me :)

Besides this does not allow us to have a non-optional line ending in the uutils that need that and it introduces the panic in the `Into<u8>` impl.

Reply:
I'm not happy with this hack neither. What do you suggest? Using `struct LineEnding(Option(u8));` would be an option?

Reply:
For the utils that need an optional line ending, they could use `Option<LineEnding>` right? Or do all utils need the optional line ending?

Reply:
Good idea, thanks! -- Done.

In fact, one once command uses an optional line ending. Thus it was simple to refactor.